docs/technical-reference/architecture.md

```diff
@@ -204,25 +203,61 @@ The argument `form` is a jQuery-wrapped FORM element that will get submitted to
+is not uncommon that this initial choice must be overriden by the user.
+
+Beyond this choice of format, the parsing UI panel offers a configuration panel for the chosen importer. This part of the UI can be defined independently for each input format, given that not all options are relevant for all formats. For instance, when
+selecting the "Text file" option, the specific UI of the `LinedBasedImporter` will be shown. This UI is defined in:
+* `main/webapp/modules/core/scripts/index/parser-interfaces/line-based-parser-ui.html`
+* `main/webapp/modules/core/scripts/index/parser-interfaces/line-based-parser-ui.js`
+
+Other importers generally define their own parsing panel as well.
+
+The link between the format's identifier (MIME type), importer (Java class which defines the parsing logic) and parsing options UI (Javascript class that defines the rendering of this options area) is made in the `main/webapp/modules/core/MOD-INF/controller.js` file,
+where those components are registered together in the `ImportingManager`.
 
 ### Server-side Components {#server-side-components}
 
 #### ImportingController {#importingcontroller}
-Documentation not currently available
 
-#### UrlRewriter {#urlrewriter}
-Documentation not currently available
+An importing controller is a component of the back-end which is in charge of the entire importing workflow, from the initial transfer of the raw data to be imported to the created project, with all the configuration steps in between, [as described in the
+earlier section](#importing-controllers). OpenRefine comes with
+a default importing controller which implements this for data coming from:
+* file upload by the user via the web interface
+* upload of textual information using the clipboard import form
+* download of a file by supplying a URL
+
+For all of these data sources, the first step consists storing the corresponding input files in a temporary directory inside the workspace. The default importing controller provides an HTTP API used by the front-end to select which files to import, [predict
+the format they are in](#formatguesser), provide default importing options for the selected format, preview the project's first few rows with the given options, and finally create the project.
+
+Extensions can define other importing controllers to implement other importing flows depending on the data source. For instance, importing data from a SQL database requires different steps such as selecting the database and providing a SQL query. The
+`database` extension implements such a workflow by providing its own importing controller.
 
 #### FormatGuesser {#formatguesser}
-Documentation not currently available
+
+A format guesser is a class that tries to determine the MIME type of a file, considering its contents.
```

Comment:
```suggestion
A format guesser is a class that tries to determine the MIME type of a file, considering its contents and/or file extension.
```
I think the above is more correct now since @tfmorris did changes on a few in the past?

Reply:
Technically speaking the FormatGuesser is only looking at the file contents: the file extension is taken into account elsewhere.

Reply:
Ah, ok.